src/coreclr/jit/lowerxarch.cpp

```diff
@@ -577,18 +577,15 @@ void Lowering::LowerPutArgStk(GenTreePutArgStk* putArgStk)
                 {
                     MakeSrcContained(putArgStk, fieldNode);
                 }
-                else if (IsContainableMemoryOp(fieldNode) && IsSafeToContainMem(putArgStk, fieldNode))
-                {
-                    MakeSrcContained(putArgStk, fieldNode);
-                }
                 else
                 {
                     // For the case where we cannot directly push the value, if we run out of registers,
                     // it would be better to defer computation until we are pushing the arguments rather
                     // than spilling, but this situation is not all that common, as most cases of FIELD_LIST
                     // are promoted structs, which do not not have a large number of fields, and of those
                     // most are lclVars or copy-propagated constants.
-                    fieldNode->SetRegOptional();
+
+                    TryMakeSrcContainedOrRegOptional(putArgStk, fieldNode);
```

Comment:
This code is tricky. What the comment talks about is that if we cannot contain the field node, we _must_ mark it register optional, "no matter what", or, to be precise, we must mark enough fields for us to not run out of available registers.

So technically, we've had a bug here by always setting setting reg optional, but not setting it would exchange that bug for another.

Not sure if any action is truly required, but it may be best to keep this code as it was.

Reply:
Reverted. Will log an issue tracking a proper fix.

Reply:
https://github.com/dotnet/runtime/issues/77901